tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -662,84 +862,329 @@ TEST(RDFVary, VaryDisplay) // TEST instead of TEST_P because Display is single-t
    // must update this test when https://github.com/root-project/root/issues/9894 is addressed
 }
 
-TEST_P(RDFVary, DISABLED_VaryFill)
-{
+struct Jet {
+   double a, b;
+};
 
-}
+struct CustomFiller {
 
-TEST_P(RDFVary, DISABLED_VaryGraph)
-{
+   TH2D h{"", "", 10, 0, 10, 10, 0, 10};
 
-}
-
-TEST_P(RDFVary, DISABLED_VaryGraphAsymmErrors)
-{
+   void Fill(const Jet &j) { h.Fill(j.a, j.b); }
 
-}
+   void Merge(const std::vector<CustomFiller *> &jets)
+   {
+      TList l;
+      for (auto *j : jets)
+         l.Add(&j->h);
+      h.Merge(&l);
+   }
+};
 
-TEST_P(RDFVary, DISABLED_VaryHisto2D)
-{
+TEST_P(RDFVary, VaryCustomObject)
+{
+   // this object can be filled
+   // Cannot extract type_info of type CLING_UNKNOWN_TYPE_Jet.
+   EXPECT_THROW(ROOT::RDataFrame(10)
+                   .Define("Jet",
+                           [] {
+                              return Jet{1., 2.};
+                           })
+                   .Vary(
+                      "Jet",
+                      [] {
+                         return ROOT::RVec<Jet>{{}, {4., 5.}};
+                      },
+                      {}, 2)
+                   .Fill<Jet>(CustomFiller{}, {"Jet"}), std::runtime_error);
```

Comment:
I'm not sure what's going on here, are you sure that "Cannot extract type_info of type CLING_UNKNOWN_TYPE_Jet" has to do with `Vary`? what's the callstack at the point the error is printed?